### ISSUE TITLE
terminate called after throwing an instance of 'std::bad_alloc', over 20Gb memory free

### ISSUE THREAD

**Issue by mullenba** [tensorflow/tensorflow#19603]
Training a small model in Keras, getting this error after about 20 epochs (of 10000 samples).  While training is running, the system is chugging along happily with about 43Gb of memory free.  At about epoch 20, suddenly everything stops and free memory drops to about 20Gb before throwing the error.

Seems to be the same issue as #9487
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

tf_env data:

> == cat /etc/issue ===============================================
> Linux 7ff559301433 4.10.0-37-generic #41-Ubuntu SMP Fri Oct 6 20:20:37 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
> VERSION="16.04.2 LTS (Xenial Xerus)"
> VERSION_ID="16.04"
> VERSION_CODENAME=xenial
> 
> == are we in docker =============================================
> Yes
> 
> == compiler =====================================================
> c++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
> Copyright (C) 2015 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> 
> 
> == uname -a =====================================================
> Linux 7ff559301433 4.10.0-37-generic #41-Ubuntu SMP Fri Oct 6 20:20:37 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
> 
> == check pips ===================================================
> numpy (1.14.3)
> protobuf (3.4.0)
> tensorflow-gpu (1.3.0)
> tensorflow-tensorboard (0.1.8)
> 
> == check for virtualenv =========================================
> False
> 
> == tensorflow import ============================================
> tf.VERSION = 1.3.0
> tf.GIT_VERSION = v1.3.0-rc2-20-g0787eee
> tf.COMPILER_VERSION = v1.3.0-rc2-20-g0787eee
> Sanity check: array([1], dtype=int32)
> 
> == env ==========================================================
> LD_LIBRARY_PATH /usr/local/nvidia/lib:/usr/local/nvidia/lib64
> DYLD_LIBRARY_PATH is unset
> 
> == nvidia-smi ===================================================
> Mon May 28 18:57:01 2018       
> +-----------------------------------------------------------------------------+
> | NVIDIA-SMI 384.81                 Driver Version: 384.81                    |
> |-------------------------------+----------------------+----------------------+
> | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
> | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
> |===============================+======================+======================|
> |   0  GeForce GTX 750 Ti  Off  | 00000000:0B:00.0 Off |                  N/A |
> | 33%   30C    P0     1W /  38W |     11MiB /  2000MiB |      0%      Default |
> +-------------------------------+----------------------+----------------------+
> |   1  GeForce GTX 660     Off  | 00000000:14:00.0 N/A |                  N/A |
> | 30%   34C    P0    N/A /  N/A |     11MiB /  1999MiB |     N/A      Default |
> +-------------------------------+----------------------+----------------------+
>                                                                                
> +-----------------------------------------------------------------------------+
> | Processes:                                                       GPU Memory |
> |  GPU       PID   Type   Process name                             Usage      |
> |=============================================================================|
> |    1                    Not Supported                                       |
> +-----------------------------------------------------------------------------+
> 
> == cuda libs  ===================================================
> /usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart.so.8.0.61
> /usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudart_static.a
> 

The data loading code isn't important, just opening files and getting 64x64 pixel chunks of each.  I'm attaching the actual NN code though.

```
import tensorflow as tf

import keras
from keras.models import Sequential
from keras.utils import multi_gpu_model
from keras.layers import Convolution2D, MaxPooling2D, Conv2DTranspose

from keras import backend as K
K.clear_session()

import numpy as np

import dataPool2 as dp

def psnr(img1, img2):
    #print(img1.shape, img2.shape)
    #assert img1.shape == img2.shape

    mse = tf.losses.mean_squared_error(img1, img2)

    score = 3 * tf.log((255 ** 2) / mse)
    
    return 1 / score

def srcnnModel():
	model = Sequential()
	model.add(Convolution2D(64, (9, 9), activation='relu', padding='same', input_shape=(64,64,1)))
	model.add(Convolution2D(32, (1, 1), activation='relu', padding='same'))
	model.add(Convolution2D(1, (5, 5), padding='same'))
	
	model.compile(loss='mse',
              optimizer='adam',
              metrics=['accuracy'])

	return model

with tf.device('/cpu:0'):
    model = srcnnModel()

print(model.summary())
	
parallel_model = multi_gpu_model(model, gpus=2)
parallel_model.compile(loss=psnr,
                       optimizer='adam')
				   
# Generate Data
xYchan, xCrCb, yYchan, origY = dp.createDataset("test_images/", 10000)

parallel_model.fit(np.array(xYchan), np.array(yYchan), epochs=200, batch_size=128, callbacks=[
                keras.callbacks.EarlyStopping(monitor='loss', min_delta=0.0001, patience=3, verbose=0, mode='auto')],) 
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by mullenba** [comment]
After further testing, changing batch size, adding/removing "clear_session()", and running in single GPU mode all have no effect on the error.

**Comment by mullenba** [comment]
More testing...

Removing "with tf.device('/cpu:0'):" allows me to see more details on the error.  I'm running out of GPU memory.  It seems crazy that a model with 8129 parameters and 5-10,000 samples won't fit in 4Gb?

**Comment by skye** [comment]
@aselle do you know who can look at this kind of issue?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by aselle** [comment]
What size tensors are you feeding and what is the size of the intermediate tensors?

**Comment by aselle** [comment]
@reedwm, do you have any further insights?

**Comment by reedwm** [comment]
Not sure what the issue could be. Perhaps @zheng-xq has some ideas?

Without a self-contained example, we cannot effectively debug this.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!

**Comment by w9ahmed** [comment]
I'm facing the same issue, I have an 8GB of RAM and an GPU with 4GB.

**Comment by ellonde** [comment]
Encounterd the same problem in TF2.0. For me it was padding='same'. Will throw error even before running graph with data

**Comment by sayakpaul** [comment]
I am also facing this same issue on my GCP VM (containing 2 V100s). I am using [this example](https://www.pyimagesearch.com/2017/10/30/how-to-multi-gpu-training-with-keras-python-and-deep-learning/) by PyImageSearch for testing. I have carefully used `tf.keras` and also made sure that I am using `tf.compat.v2.keras.utils.multi_gpu_model`. Here's some error trace:

```
2019-11-19 03:07:59.945565: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device i
nterconnect StreamExecutor with strength 1 edge matrix:
2019-11-19 03:07:59.945624: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]      0 1
 
2019-11-19 03:07:59.945680: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 0:   N Y
 
2019-11-19 03:07:59.945736: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1178] 1:   Y N
 
2019-11-19 03:07:59.946193: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1006] succ
essful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA
 node, so returning NUMA node zero
2019-11-19 03:07:59.946714: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1006] succ
essful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA
 node, so returning NUMA node zero
2019-11-19 03:07:59.947233: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1006] succ
essful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA
 node, so returning NUMA node zero
2019-11-19 03:07:59.947712: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created 
TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14828 MB memory) -> phys
ical GPU (device: 0, name: Tesla V100-SXM2-16GB, pci bus id: 0000:00:04.0, compute capability
: 7.0)
2019-11-19 03:07:59.947868: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1006] succ
essful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA
 node, so returning NUMA node zero
2019-11-19 03:07:59.948324: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1304] Created 
TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 14828 MB memory) -> phys
ical GPU (device: 1, name: Tesla V100-SXM2-16GB, pci bus id: 0000:00:05.0, compute capability
: 7.0)
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted
```
